packages/next/lib/check-lock-file.ts

```diff
@@ -0,0 +1,23 @@
+import lock from 'proper-lockfile'
+import { mkdirSync } from 'fs'
+
+export default function checkLockFile(distDir: string) {
+  try {
+    mkdirSync(distDir)
+  } catch (err) {
+    if (err.code !== 'EEXIST') {
+      throw err
```

Comment:
I don't think we want this to be fatal. Let's just warn and proceed without lock IMO.

Reply:
It will still be fatal since we can't build if we don't have a `distDir` it will just throw after this step